app/models/moneyservicebusiness/MoneyServiceBusiness.scala

```diff
@@ -41,40 +41,41 @@ case class MoneyServiceBusiness(
 
   def businessUseAnIPSP(p: BusinessUseAnIPSP): MoneyServiceBusiness =
-    this.copy(businessUseAnIPSP = Some(p), hasChanged = hasChanged || !this.businessUseAnIPSP.contains(p), hasAccepted = this.businessUseAnIPSP.contains(p))
+    this.copy(businessUseAnIPSP = Some(p), hasChanged = hasChanged || !this.businessUseAnIPSP.contains(p), hasAccepted = hasAccepted && this.businessUseAnIPSP.contains(p))
 
   def identifyLinkedTransactions(p: IdentifyLinkedTransactions): MoneyServiceBusiness =
-    this.copy(identifyLinkedTransactions = Some(p), hasChanged = hasChanged || !this.identifyLinkedTransactions.contains(p), hasAccepted = this.identifyLinkedTransactions.contains(p))
+    this.copy(identifyLinkedTransactions = Some(p), hasChanged = hasChanged || !this.identifyLinkedTransactions.contains(p), hasAccepted = hasAccepted && this.identifyLinkedTransactions.contains(p))
 
   def fundsTransfer(p: FundsTransfer): MoneyServiceBusiness =
-    this.copy(fundsTransfer = Some(p), hasChanged = hasChanged || !this.fundsTransfer.contains(p), hasAccepted = this.fundsTransfer.contains(p))
+    this.copy(fundsTransfer = Some(p), hasChanged = hasChanged || !this.fundsTransfer.contains(p), hasAccepted = hasAccepted && this.fundsTransfer.contains(p))
 
   def branchesOrAgents(p: BranchesOrAgents): MoneyServiceBusiness =
-    this.copy(branchesOrAgents = Some(p), hasChanged = hasChanged || !this.branchesOrAgents.contains(p), hasAccepted = this.branchesOrAgents.contains(p))
+    this.copy(branchesOrAgents = Some(p), hasChanged = hasChanged || !this.branchesOrAgents.contains(p), hasAccepted = hasAccepted && this.branchesOrAgents.contains(p))
 
   def sendMoneyToOtherCountry(p: SendMoneyToOtherCountry): MoneyServiceBusiness =
-    this.copy(sendMoneyToOtherCountry = Some(p), hasChanged = hasChanged || !this.sendMoneyToOtherCountry.contains(p), hasAccepted = this.sendMoneyToOtherCountry.contains(p))
+    this.copy(sendMoneyToOtherCountry = Some(p), hasChanged = hasChanged || !this.sendMoneyToOtherCountry.contains(p), hasAccepted = hasAccepted && this.sendMoneyToOtherCountry.contains(p))
 
-  def sendTheLargestAmountsOfMoney(p: SendTheLargestAmountsOfMoney): MoneyServiceBusiness =
-    this.copy(sendTheLargestAmountsOfMoney = Some(p), hasChanged = hasChanged || !this.sendTheLargestAmountsOfMoney.contains(p), hasAccepted = this.sendTheLargestAmountsOfMoney.contains(p))
+  def sendTheLargestAmountsOfMoney(p: Option[SendTheLargestAmountsOfMoney]): MoneyServiceBusiness =
+    this.copy(sendTheLargestAmountsOfMoney = p, hasChanged = hasChanged || !this.sendTheLargestAmountsOfMoney.equals(p),
+      hasAccepted = hasAccepted && this.sendTheLargestAmountsOfMoney.equals(p))
 
-  def mostTransactions(p: MostTransactions): MoneyServiceBusiness =
-    this.copy(mostTransactions = Some(p), hasChanged = hasChanged || !this.mostTransactions.contains(p), hasAccepted = this.mostTransactions.contains(p))
+  def mostTransactions(p: Option[MostTransactions]): MoneyServiceBusiness =
```

Comment:
Why change these to options?

Reply:
I need to be able to clear down both largest amount and most transactions. This required me to pass in `none` at the point of clearing them down. If the params are not options then I can't pass in `None`. 

```
   case _ => val newModel = msb
                    .sendMoneyToOtherCountry(data)
                    .sendTheLargestAmountsOfMoney(None)
                    .mostTransactions(None)

                    dataCacheConnector.save(MoneyServiceBusiness.key, newModel) map {
                    _ => routing(data.money, services.msbServices,register, newModel, edit)
```